Diagnostics/HealthChecker/Analyzer/Invoke-AnalyzerOsInformation.ps1

```diff
@@ -96,68 +96,107 @@ Function Invoke-AnalyzerOsInformation {
+            Write-Verbose "Configured PageFile detected"
+            $displayValue = ($pageFileDisplayTemplate -f $pageFileObj.Name, $pageFileObj.MaxPageSize)
+        }
 
-            if ($recommendedPageFile -ne $maxPageSize) {
-                $displayValue = "$maxPageSize`MB `r`n`t`tWarning: Page File is not set to 25% of the Total System Memory which is $totalPhysicalMemory`MB. Recommended is $recommendedPageFile`MB"
+        if ($exchangeInformation.BuildInformation.MajorVersion -eq [HealthChecker.ExchangeMajorVersion]::Exchange2019) {
+            $recommendedPageFile = [Math]::Round($totalPhysicalMemory / 4)
+            $pageFileObj.RecommendedPageFile = $recommendedPageFile
+            Write-Verbose "System is running Exchange 2019. Recommended PageFile Size: $recommendedPageFile"
+
+            $recommendedPageFileWording2019 = "On Exchange 2019, the recommended PageFile size is 25% ({0}MB) of the total system memory ({1}MB)."
+            if ($pageFileObj.MaxPageSize -eq 0) {
+                $pageFileAdditionalDisplayValue = ("Error: $recommendedPageFileWording2019" -f $recommendedPageFile, $totalPhysicalMemory)
+            } elseif ($recommendedPageFile -ne $pageFileObj.MaxPageSize) {
+                $pageFileAdditionalDisplayValue = ("Warning: $recommendedPageFileWording2019" -f $recommendedPageFile, $totalPhysicalMemory)
             } else {
-                $displayValue = "$recommendedPageFile`MB"
+                $displayValue = ($pageFileDisplayTemplate -f $pageFileObj.Name, $recommendedPageFile)
                 $displayWriteType = "Grey"
             }
         } elseif ($totalPhysicalMemory -ge 32768) {
-            if ($maxPageSize -eq 32778) {
-                $displayValue = "$maxPageSize`MB"
+            Write-Verbose "System is not running Exchange 2019 and has more than 32GB memory. Recommended PageFile Size: 32778MB"
+
+            $recommendedPageFileWording32GBPlus = "PageFile should be capped at 32778MB for 32GB plus 10MB."
+            if ($pageFileObj.MaxPageSize -eq 0) {
+                $pageFileAdditionalDisplayValue = "Error: $recommendedPageFileWording32GBPlus"
+            } elseif ($pageFileObj.MaxPageSize -eq 32778) {
                 $displayWriteType = "Grey"
             } else {
-                $displayValue = "$maxPageSize`MB `r`n`t`tWarning: Pagefile should be capped at 32778MB for 32GB plus 10MB - Article: https://aka.ms/HC-SystemRequirements2016#hardware-requirements-for-exchange-2016"
+                $pageFileAdditionalDisplayValue = "Warning: $recommendedPageFileWording32GBPlus"
             }
         } else {
             $recommendedPageFile = $totalPhysicalMemory + 10
-            $testingValue.RecommendedPageFile
-
-            if ($recommendedPageFile -ne $maxPageSize) {
-                $displayValue = "$maxPageSize`MB `r`n`t`tWarning: Page File is not set to Total System Memory plus 10MB which should be $recommendedPageFile`MB"
+            $pageFileObj.RecommendedPageFile
```

Comment:
Not sure why this is here. We are just displaying the value to the console, this shouldn't happen.

Reply:
Right, pasted something by mistake. Fixed now.